### PR TITLE
Allow initial ID portions for cluster lookups

### PR DESCRIPTION
**Pull request by bamarni**
It makes the CLI more user-friendly.

This is also similar to how the `docker` CLI works with container names / IDs.

### CODE REVIEW

**Comment by bamarni** [comment]
run linux integration tests

**Comment by bamarni** [comment]
run linux integration tests

**Comment by bamarni** [comment]
run linux integration tests

**Comment by bamarni** [comment]
run all integration tests

**Comment by bamarni** [comment]
run all integration tests

File: dcos/cluster.py [hunk 1]
```diff
@@ -276,12 +276,20 @@ def get_cluster(name):
     """
     :param name: name of cluster
     :type name: str
-    :returns: Cluster identified by name
+    :returns: Cluster identified by name or ID, it also accepts an intial
```
Comment (ArmandGrillet) [inline dcos/cluster.py]: s/`it also accepts an intial`/`also accepts an initial`.

File: dcos/cluster.py [hunk 2]
```diff
@@ -276,12 +276,20 @@ def get_cluster(name):
...
+              portion of ID if it's unique.
```
Comment (ArmandGrillet) [inline dcos/cluster.py]: Is it the same in Docker? It sounds quite strange to match portion of IDs.

File: dcos/cluster.py [hunk 3]
```diff
@@ -276,12 +276,20 @@ def get_cluster(name):
...
     """
 
-    return next((c for c in get_clusters(True)
-                 if c.get_cluster_id() == name or c.get_name() == name), None)
+    clusters = [c for c in get_clusters(True)
+                if c.get_name() == name
+                or c.get_cluster_id().startswith(name)]
+
+    if len(clusters) > 1:
+        msg = 'Multiple clusters matching "{}", please use the cluster ID.'
```
Comment (ArmandGrillet) [inline dcos/cluster.py]: s/`please use the cluster ID.`/`please use the exact cluster ID.`.

**Comment by bamarni** [comment]
run linux integration tests

File: dcos/cluster.py (hunk 2 above)
Comment (bamarni) [reply dcos/cluster.py]: it only work with the initial portion of ID, and if it's the unique match.

**Comment by bamarni** [comment]
run linux integration tests

**Pull request merged**